modules/auth/sso/sspi_windows.go

```diff
@@ -0,0 +1,307 @@
+// without authentication (eg. the Web App Manifest, the Service Worker script or the favicon)
+func isPublicResource(ctx *macaron.Context) bool {
+	path := strings.TrimSuffix(ctx.Req.URL.Path, "/")
+	return path == "/robots.txt" ||
+		path == "/favicon.ico" ||
+		path == "/favicon.png" ||
+		path == "/manifest.json" ||
+		path == "/serviceworker.js"
+}
+
+// isPublicPage checks if the url is of a public page that should not require authentication
+func isPublicPage(ctx *macaron.Context) bool {
+	path := strings.TrimSuffix(ctx.Req.URL.Path, "/")
+	homePage := strings.TrimSuffix(setting.AppSubURL, "/")
+	currentURL := homePage + path
+	return currentURL == homePage ||
+		path == "/user/login" ||
+		path == "/user/login/openid" ||
+		path == "/user/sign_up" ||
+		path == "/user/forgot_password" ||
+		path == "/user/openid/connect" ||
+		path == "/user/openid/register" ||
+		strings.HasPrefix(path, "/user/oauth2") ||
+		path == "/user/link_account" ||
+		path == "/user/link_account_signin" ||
+		path == "/user/link_account_signup" ||
+		path == "/user/two_factor" ||
+		path == "/user/two_factor/scratch" ||
+		path == "/user/u2f" ||
+		path == "/user/u2f/challenge" ||
+		path == "/user/u2f/sign" ||
+		(!setting.Service.RequireSignInView && (path == "/explore/repos" ||
+			path == "/explore/users" ||
+			path == "/explore/organizations" ||
+			path == "/explore/code"))
+}
+
+// stripDomainNames removes NETBIOS domain name and separator from down-level logon names
+// (eg. "DOMAIN\user" becomes "user"), and removes the UPN suffix (domain name) and separator
+// from UPNs (eg. "user@domain.local" becomes "user")
+func stripDomainNames(username string) string {
+	if strings.Contains(username, "\\") {
+		parts := strings.SplitN(username, "\\", 2)
+		if len(parts) > 1 {
+			username = parts[1]
+		}
+	} else if strings.Contains(username, "@") {
+		parts := strings.Split(username, "@")
+		if len(parts) > 1 {
+			username = parts[0]
+		}
+	}
+	return username
+}
+
+func replaceSeparators(username string, cfg *models.SSPIConfig) string {
+	newSep := cfg.SeparatorReplacement
+	username = strings.ReplaceAll(username, "\\", newSep)
+	username = strings.ReplaceAll(username, "/", newSep)
+	username = strings.ReplaceAll(username, "@", newSep)
+	return username
+}
+
+func sanitizeUsername(username string, cfg *models.SSPIConfig) string {
+	if len(username) == 0 {
+		return ""
+	}
+	if cfg.StripDomainNames {
+		username = stripDomainNames(username)
+	}
+	// Replace separators even if we have already stripped the domain name part,
+	// as the username can contain several separators: eg. "MICROSOFT\useremail@live.com"
+	username = replaceSeparators(username, cfg)
+	return username
+}
+
+// handleSignIn clears existing session variables and stores new ones for the specified user object
+func handleSignIn(ctx *macaron.Context, sess session.Store, user *models.User) {
+	_ = sess.Delete("openid_verified_uri")
```

Comment:
It seems to me that this should also be moved out of sspi_windows.go

Reply:
Moved out of `sspi_windows.go` into `sso.go`, by commit https://github.com/go-gitea/gitea/pull/8463/commits/533c1a79482440b04d2e61b1b21599f9a0edba61